### ISSUE TITLE
Conflict two actually not a conflict moves

### ISSUE THREAD

**Issue by statiolake** [takashi-chikayama/SamurAI-Dig-Here-2020-21#4]
I'm using current master (4fdda069bbcd345d71b56d3708ad997fe3623d37) but there still seems to be some bugs in both manager and visualizer. I'm sorry if you already know this issue and are working on.

The problematic JSON is here: <https://gist.github.com/statiolake/ca8a1470a3fbdd4c0e0efb90f1642325>

Loading this JSON to the visualizer gives me an error `Position of agent 3 does not match at step 3: Recorded: (6,3) Actual: (7,4)`.

For this step, manager log is:

```
(... snip ...)
Step: 3
Golds known: 18@(8,9) 18@(2,5) 12@(1,0) 6@(4,1) 10@(7,8) 6@(0,4) 6@(5,5) 10@(1,3) 4@(9,2) 6@(0,7) 20@(2,8) 2@(3,1) 18@(2,4) 8@(6,3) 10@(8,4) 20@(6,4)
Agent 0@(3,6) stay (3,6)  59647 msec left
Agent 1@(8,7) plug (8,8)  59708 msec left
Agent 2@(6,4) move (7,5)X 59658 msec left; Crossing move lines
Agent 3@(7,4) move (6,3)  59597 msec left
Scores are 10:0
(... snip ...)
```

The log says the _red_ dog's move is unacceptable for the reason of crossing move line, but actually this is not. On the other hand, visualizer is complaining about the position of _blue_ dog and saying the blue dog should stay (7,4).

Their plan can be illustrated like this.

![image](https://user-images.githubusercontent.com/20490597/103686370-104adc00-4fd2-11eb-923b-d840a18867fd.png)

In my understanding this is not a conflict.

Note: With some more experiment, I'm not sure, but it seems the visualizer is considering both red and blue dog's moves should be rejected.

**Comment by task-h** [comment]
Thank you for your report. We expect the issues #3 and #4 are solved in the latest (481c9b2) version.

**Comment by statiolake** [comment]
Thanks! Both seems to be fixed.
